internal/controller/account/entitlement/entitlement_test.go

```diff
@@ -90,6 +90,43 @@ func TestObserve(t *testing.T) {
 				err: errors.New(errKubeAPI),
 			},
 		},
+		"Simple Case, no unique identifier passed, api error for ambigous names": {
+			args: args{
+				kube: &test.MockClient{
+					MockStatusUpdate: noopStatusUpdate, MockList: test.NewMockListFn(nil, ListEntitlements(entitlement(withServiceName("hana-cloud"), withUniqueServicePlanIdentifier("a")), entitlement(withServiceName("hana-cloud"), withUniqueServicePlanIdentifier("b")))),
+				},
+				client: fake.MockClient{MockDescribeCluster: func(ctx context.Context, input v1alpha1.Entitlement) (*entitlement2.Instance, error) {
+					return nil, errors.New(errClientAPI)
```

Comment:
This is not testing the right logic. 
Basically you just verify here that "if error comes back from client, return it from controller", which is the same as the test case above.

Since you test the internal logic in the client, I guess its find to not have additional tests here. The test cases here cover more of the interaction with other components.

Reply:
Perfect would be to test the Create to make sure this new attribute is passed to the api client. But since we do not have any tests other then observe yet that would be more effort, so from my side no need to do that right now.